proto/voice-info.proto

```diff
@@ -0,0 +1,175 @@
+    FORMAT_CPIO = 2;            // cpio archive
+  }
+  string path = 1;                // relative path, mandatory
+  Compression compression = 2;    // file compression, optional
+  Format format = 3;              // file format, optional
+  string md5 = 4;                 // MD5 sum of file, mandatory
+}
+
+/* Speech model language / locale info */
+message Language {
+  /* <ISO 639 language code>_<ISO 3166 country code> */
+  enum Locale {
+    /* default: any language/country) */
+    LOCALE_ANY = 0;
+    is_IS = 1;
+    /* more entries can be appended here ... */
+  }
+  Locale locale = 1;  // locale of the voice, mandatory
+  string name = 2;    // language name of the voice, mandatory
+}
+
+/* Phonemes used for acoustic model, alphabets and symbol mappings */
+message PhonemeDescription {
+  enum PhonemeType {
+    PHONEME_TYPE_INVALID = 0;
+    /* use IPA symbols, these are always distinct single symbols */
+    PHONEME_TYPE_IPA = 1;
+    /* use "extended Sampas", these are mostly made up of multiple symbols */
+    PHONEME_TYPE_XSAMPA = 2;
+  }
+  /* Mapping between Alphabet and model representation */
+  message AlphabetMapping {
+    /*
+      Mapping of all IPA/XSampa symbols to their corresponding model value,
+      as well as meta-symbols like <sp> "short pause", <pau> "pause", etc. used
+      in model training.
+    */
+    map<string, uint32> symbols = 1;
+  }
+  PhonemeType phoneme_type = 1; // phoneme type, mandatory
+  AlphabetMapping mapping = 2;  // symbol mappings, mandatory
+  bool syllabified = 3;         // "." at each syllable boundary, optional
+  bool stress_labels = 4;       // "0" or "1" after each vowel, optional
+  string word_separator = 5;    // word separator; default "-", optional
+}
+
+/* For NN models: used neural network format */
+enum FormatNN {
+  NN_FORMAT_INVALID = 0;
+  NN_FORMAT_TORCH = 1;          // plain pyTorch
+  NN_FORMAT_TORCHSCRIPT = 2;    // Torchscript
+  NN_FORMAT_TORCH_MOBILE = 3;   // Torchscript mobile
+  NN_FORMAT_ONNX = 4;           // ONNX
+  NN_FORMAT_NNAPI = 5;          // Android NNAPI
+}
+
+/* Voice acoustic model description */
+message AcousticModel {
+  /* Type of the acoustic model */
+  enum AcousticModelType {
+    ACOUSTIC_MODEL_INVALID = 0;
+    ACOUSTIC_MODEL_FESTIVAL_UNIT_SELECTION = 1;
+    ACOUSTIC_MODEL_FESTIVAL_FLITE_GRAPHEME = 2;
+    ACOUSTIC_MODEL_FASTSPEECH2 = 3;
+    ACOUSTIC_MODEL_FASTSPEECH2_ESPNET = 4;
+  }
+  
+  AcousticModelType type = 1;     // acoustic model type
+  FormatNN nn_format = 2;         // for NN models: used neural network format
+  PhonemeDescription phonemes = 3;// describe phonemes and model mappings,
+                                  // mandatory
+  string description = 4;         // acoustic model description, mandatory
+  FileEntry file = 5;             // acoustic model file specifications,
+                                  // mandatory
+  Revision revision = 6;          // acoustic model revision, mandatory
```

Comment:
any more meta data necessary ? What about original sample rate used, bit depth, etc ?

Reply:
To describe the compatibility of an am and a vocoder we would need a description of the features that the am outputs and that the vocoder accepts. That would include feature type (e.g. `mfcc`, `fbank`), `fft_size`, `freq_min`, `freq_max`, `feat_dim`/`num_bins` and `sample_rate`.

Reply:
could you sketch out these parameters and their types ?